Docs/reference/content/reference/driver/crud/compression.md

```diff
@@ -0,0 +1,109 @@
++++
+date = "2020-01-29T22:05:03-04:00"
+title = "Compression"
+[menu.main]
+  parent = "Reference Reading and Writing"
+  identifier = "Compression"
+  weight = 25
+  pre = "<i class='fa'></i>"
++++
+
+## Compression
+
+The C# driver supports compression of messages to and from MongoDB servers. The driver implements the three algorithms that are supported by MongoDB servers:
+
+* [Snappy](https://google.github.io/snappy/): Snappy compression can be used when connecting to MongoDB servers starting with the 3.4 
+release.
+* [Zlib](https://zlib.net/): Zlib compression can be used when connecting to MongoDB servers starting with the 3.6 release.
+* [Zstandard](https://facebook.github.io/zstd/): Zstandard compression can be used when connecting to MongoDB servers starting with the 4.2 release.
+
+The driver will negotiate which, if any, compression algorithm is used based on capabilities advertised by the server in the [ismaster]({{<docsref "reference/command/isMaster/">}}) command response. 
+
+### Specify compression via `URI`
+
+```c#
+using MongoDB.Driver;
+```
+
+To specify compression with [`ConnectionString`]({{<apiref "T_MongoDB_Driver_Core_Configuration_ConnectionString">}}), just add `compressors` into the connection string, as in:
+
+```c#
+var mongoUrl = new MongoUrl("mongodb://localhost/?compressors=snappy");
+var client = new MongoClient(mongoUrl);
+```
+for Snappy compression, or
+
+```c#
+var mongoUrl = new MongoUrl("mongodb://localhost/?compressors=zlib");
```

Comment:
Java doc doesn't say anything about `zlibcompressionlevel`. Should I mention this option?

Reply:
Sure, you can mention it.

Reply:
done